### PR TITLE
Base scale: add baseline feature, add clamping option, refactor a bit…

### PR DESCRIPTION
**Pull request by angieskazka**
… for readability. Size scale: remove square root (should be done in component), configure for clamping and zero baseline

### CODE REVIEW

File: src/core/scale/baseScale.js [hunk 1]
```diff
@@ -77,9 +85,21 @@ export function baseScale(config = {}, parent) {
...
-            return this.config.domain ? this.config.domain.map(c => parseConfigValue(c, this.data.conceptProps))
-                : this.data.domain ? this.data.domain
-                : defaults.domain;
+            if (this.config.domain) {
+                return this.config.domain.map(c => parseConfigValue(c, this.data.conceptProps));
+            } else if (this.data.domain) {   
+                //fixBaseline can override the domain if defined and if data domain is one-sided
+                //by replaceing the value closest to zero with the one provided in config
+                //use cases: forcing zero-based bar charts and bubble size
+                if (this.config.fixBaseline != null && isArrayOneSided(this.data.domain)){
```
Comment (jheeffer) [inline src/core/scale/baseScale.js:94]: If `baseline <> 0`, `isArrayOneSided` should use that baseline for check right?
And your "closestToZero" doesn't work either for non-0 baseline.
So yeah, let's make it zero for now until we find other use cases. Easier.

Comment (jheeffer) [reply src/core/scale/baseScale.js:94]: Should this check if scale is continuous or at least a numeric domain? Is there a use case for zero base line with discrete numeric domains?

File: src/core/scale/baseScale.js [hunk 2]
```diff
@@ -77,9 +85,21 @@ export function baseScale(config = {}, parent) {
...
+                  let closestToZero = d3.scan(this.data.domain.map(d => Math.abs(d)));
+                  return this.data.domain.map((d,i)=>(i == closestToZero ? this.config.fixBaseline : d));
```
Comment (jheeffer) [inline src/core/scale/baseScale.js:96]:
```
const domain = [...this.data.domain];
const closestToZeroIdx = d3.scan(domain.map(Math.abs));
domain[closestToZeroIdx] = 0;
return domain;
```
maybe longer, but simpler?

File: src/core/scale/baseScale.js [hunk 3]
```diff
@@ -99,7 +119,7 @@ export function baseScale(config = {}, parent) {
                 //TODO
                 //scale.constant(limitsObj.minAbsNear0);
             }
-                
+            if(scale.clamp) scale.clamp(this.config.clamp);
```
[the remark this answers is not in the document]
Comment (angieskazka) [reply src/core/scale/baseScale.js:122]: ✅

**Pull request merged**